interfaces.md

```diff
@@ -0,0 +1,195 @@
+  output?: ITemplateOutput[] // pointers to created items/groups, with refs back to templates
+}
+```
+
+**PROPOSED**
+This separated the Item and Group templates, simplifying the Processor interfaces.
+This will require a schema transformation for existing solution items.
+
+```js
+interface ISolutionData {
+  itemTemplates: IItemTemplate[];
+  groupTemplate: IGroupTemplate[];
+  // metadata: object; // This is currently in the Solution items, but does not seem to be used...
+  params: object; // parameters needed for the solution - aka indicators
+  output?: ITemplateOutput[] // pointers to created items/groups, with refs back to templates
+}
+```
+
+
+## TemplateType
+**Note**: Not sure if we really need this... we could move the `type` property up one level and it's an IGroupTemplate when `.type === "Group"`. Seems less typescripty, but this seems overkillish for how rare Groups are.
+
+```js
+type TemplateType = "ITEM" || "GROUP";
+```
+
+## ITemplate
+Base for IItemTemplate and IGroupTemplate
+
+```js
+interface ITemplate {
+  id: string; 
+  resources: ISolutionResource[] // array of resources associated with this template
+}
+```
+
+## IGroupTemplate extends ITemplate
+Details for groups that need to be created. Stored in `data.groupTemplates` array
+```js
+interface IGroupTemplate extends ITemplate {
+  group: object; // group properties with {{variables}} injected if needed
+  itemsToShare: string[]; // id of the items to be shared to this group when they are created
+}
+```
+
+## IItemTemplate extends ITemplate
+This is an entry in the solution's `data.itemTemplates` array
+
+```js
+interface IItemTemplate extends ITemplate {
+  type: string; // item type
+  item: object; // basically an IItem w/o it's Id, and with {{variables}} injected
+  data: object ;// the /data of the item
+  properties: object; // other type-specific information, could be form, layer schema etc
+  dependencies: IDependency[]; // array referencing other items/groups this item depends on
+  relatedItems: IRelatedItems[]; // relationship information so we can re-connect during deployment
+  groupsToShareWith: string[]; // array of groupids the resulting item should be shared to
+}
+```
+
+## ISolutionResource
+
+When 'in-memory' the sourceUrl's point to the original resource location. When in a Solution Template item, the sourceUrl will point to resources of the Solution Template item.
+
+```js
+interface ISolutionResource {
+  fileName: string; // file name of the resource itself
+  type: ResourceType; // resoruce type information
+  path: string; // folder path used on destination item
+  sourceUrl: string; // url where this can be fetched from
+}
+```
+
+### ResourceType
+**TODO**: Check with Mike T re: when `fakezip` is used, and how we would handle that for the in-memory scenario
+
+```js
+type SolutionResourceType = "thumbnail" | "metadata" | "resource" | "data" | "fakezip"
+```
+
+
+## IDependency
+Returned 
```

Comment:
what is this model used for?

Reply:
Yeah... I added it and as you can see... I don't recall why/where I intended to use it... will remove